src/lib.rs

```diff
@@ -117,16 +117,24 @@ impl BarcodeScanner {
 		loop {
 			let events = self.device.fetch_events()
 				.map_err(|e| Error::new(format!("Failed to fetch events from input device: {e}")))?;
+
+			let mut shift_pressed = false;
 			for event in events {
 				// Check if key is pressed (value 1 for the key pressed, velue 0 for the key released).
 				if event.event_type() == evdev::EventType::KEY && event.value() == 1 {
 					// Create Key object based on the code.
 					let key_name = evdev::Key(event.code());
 
-					// Map key_name to the number or char.
-					if let Some(c) = key_to_str(key_name) {
-						self.buffer.push(c);
-					}
+					if key_name == evdev::Key::KEY_LEFTSHIFT {
```

Comment:
I added support for right shift and caps lock. The led is something that we may turn on when we handle the caps lock key. It's not something the keyboard does by itself. Since it's a scanner, I don't think there is need for that.